clients/admin-ui/src/features/system/history/helpers.tsx

```diff
@@ -245,3 +266,96 @@ export const alignPrivacyDeclarations = (
     },
   };
 };
+
+/** Makes sure the before and after custom_field objects have the same keys
+ * to align the rendering of the fields in the diff modal */
+export const alignSystemCustomFields = (
+  history: SystemHistoryResponse
+): SystemHistoryResponse => {
+  const beforeCustomFields = { ...history.before.custom_fields };
+  const afterCustomFields = { ...history.after.custom_fields };
+
+  const allKeys = new Set([
+    ...Object.keys(beforeCustomFields),
+    ...Object.keys(afterCustomFields),
+  ]);
+
+  allKeys.forEach((key) => {
+    if (!(key in beforeCustomFields)) {
+      beforeCustomFields[key] = null;
+    }
+    if (!(key in afterCustomFields)) {
+      afterCustomFields[key] = null;
+    }
+  });
```

Comment:
ah okay—I think this is the problem with the alignment. if I have a `beforeCustomFields` that has only one field, but my `afterCustomFields` has two fields...

```
before: {a: 'field'}
after: {b: 'field', a: 'field'}
```
then we do this `forEach` and that makes it...

```
before: {a: 'field', b: 'null}
after: {b: 'field', a: 'field'}
```

and then we get a misalignment. this is tricky—it appears JS objects [don't guarantee sort order](https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order). so we may need to sort somewhere else, or create this object a little differently

Reply:
perhaps you can recreate the object entirely, i.e. get all the keys, then iterate and not just fill in nulls—fill in the real values too. something like... (this is untested!)

```javascript
const before = Object.fromEntries(allKeys.map(key => { 
  [key, history.before.customFields[key] ?? null]
}))
const after = Object.fromEntries(allKeys.map(key => { 
  [key, history.after.customFields[key] ?? null]
}))
```